core/blockchain.go

```diff
@@ -1936,6 +1996,12 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 	)
 	parent := it.previous()
 	for parent != nil && !bc.HasState(parent.Root) {
+		if bc.StateRecoverable(parent.Root) {
```

Comment:
StateRecoverable also seems to self-contain HasState. Could we perhaps somehow simplify or remove that check here?

Reply:
yep, should be feasible